Table layout improvements

tlrobinson opened the pull request:
Resolves most of #4621 ("Up our table layout game") and a couple things in #4913 ("Drill Polish 3"):

1. tighten up row height #4621
2. right align numbers #4621
3. don't show sort arrow on hover (only if column is sorted) #4913
4. make header click target full cell #4913

tlrobinson committed:
Table: Right align numbers, tighten up spacing, fix scroll bar hiding

tlrobinson committed:
Make click target + cursor-pointer in header cells consistent

tlrobinson commented:
Also I haven't implemented these changes for the dashboard version of tables yet. If this looks good I'll do that.

attekei commented on frontend/src/metabase/lib/formatting.js:
```diff
@@ -64,20 +78,54 @@ function formatMajorMinor(major, minor, options = {}) {
+    // Tooltips should show full month name, but condense "MMMM D, YYYY - MMMM D, YYYY" to "MMMM D - D, YYYY" etc
+    const monthFormat = options.type === "tooltip" ? "MMMM" : "MMM";
+    const condensed = options.type === "tooltip";
+    // use en dashes, for Maz
```
😄

tlrobinson replied:
Hmm it does add an additional DOM node per cell, that's the only reason I can think of it being less smooth. That seemed necessary to get both the contents aligned vertically, and the ellipsis behavior.

We don't need to use flex to align it vertically, it just seemed cleaner than picking magic padding sizes.

attekei reviewed (commented):
I really like the new table layout! My only concern is that now scrolling the table is less smooth than before. Do you know if it's easy to improve the scrolling performance?

I also tested that the new week numbering seems to work well.

tlrobinson commented:
Hmm it does add an additional DOM node per cell, that's the only reason I can think of it being less smooth. That seemed necessary to get both the contents aligned vertically, and the ellipsis behavior.

We don't need to use flex to align it vertically, it just seemed cleaner than picking magic padding sizes.

attekei commented:
I think that the difference is smoothness is mostly caused by higher density of table cells, so probably the extra DOM node per cell isn't such a big factor.

tlrobinson commented:
@metabase/core-developers mind trying this out and letting me know if you notice any worse scroll performace?

mazameli reviewed (approved):
LGTM!

tlrobinson committed:
Fix flow error

salsakran commented:
I'm still seeing the sort icon on hover.

Also, definitely getting more lag on scrolling.

tlrobinson commented:
@attekei @salsakran Do you mind reverting just the ROW_HEIGHT constant (https://github.com/metabase/metabase/pull/4977/files#diff-fb164def68c92e5830ecc8ac0f165a2aR25) and seeing if that fixes the scroll lagginess for you? That will tell us if it's the additional DOM nodes or additional row density.

tlrobinson commented:
@kdoh @attekei and ideas how to avoid introducing the extra DOM node and still use flex align-center + ellipsification?

attekei commented:
My first idea is to get rid of that `<span>` in https://github.com/metabase/metabase/blob/6f13cbdbcfbabfcfcb7786589a284e988778fc1a/frontend/src/metabase/components/Value.jsx#L19, what do you think? It could already be a considerable perf boost.

tlrobinson commented:
That's there because `formatValue` might return a string which isn't a valid React element.

I suppose we could just use `formatValue` directly instead of the `Value` component.

tlrobinson committed:
Merge branch 'master' of github.com:metabase/metabase into tablepalooza

tlrobinson committed:
Merge branch 'release-0.24.0' of github.com:metabase/metabase into tablepalooza

tlrobinson committed:
Use formatValue directly instead of <Value> for slightly improved scroll performance

tlrobinson commented:
I can't repro "I'm still seeing the sort icon on hover.", can anyone else?

tlrobinson committed:
Merge branch 'master' of github.com:metabase/metabase into tablepalooza

tlrobinson committed:
Right align numbers on TableSimple

tlrobinson committed:
Merge branch 'master' of github.com:metabase/metabase into tablepalooza

tlrobinson commented:
Ok I removed the extraneous <span> and it feels slightly faster, but I do think the majority of the performance regression is just due to the higher density of cells.

I've also implemented right alignment on the dashboard version of tables, so I think this is ready for review.

attekei reviewed (commented):
Yep, it feels faster (hopefully it's not all placebo 😉), but I'm not sure if the current density is good or if the row height should be increased a little.

One other minor question: should Lat/Lon values also be right-aligned?

tlrobinson committed:
Merge branch 'master' of github.com:metabase/metabase into tablepalooza

tlrobinson committed:
Increase table row height slightly

tlrobinson committed:
Right align lat/lon. Add tests

tlrobinson commented:
I agree it feels slightly too dense. I've increased the row height from 28px:

![screenshot 2017-05-19 14 06 35](https://cloud.githubusercontent.com/assets/18193/26266924/74c21a24-3c9c-11e7-8a73-171b354865c0.png)

to 30px:

![screenshot 2017-05-19 14 06 16](https://cloud.githubusercontent.com/assets/18193/26266931/78ad4e60-3c9c-11e7-82ac-df7c94ee0af2.png)

I've also included lat/lon in the right aligned fields.

tlrobinson commented:
This should be ready to go, or at least another round of review.

salsakran commented:
feels reasonable to me.

at this point the grid still irritates me, but right aligning stuff is really nice.

mazameli commented:
As they say in the freight industry, "ship it."
